src/atom_vec.cpp

```diff
@@ -2572,13 +2562,13 @@ void AtomVec::create_method(int nfield, Method *method)
 
 void AtomVec::init_method(Method *method)
 {
-  method->pdata = NULL;
-  method->datatype = NULL;
-  method->cols = NULL;
-  method->maxcols = NULL;
-  method->collength = NULL;
-  method->plength = NULL;
-  method->index = NULL;
+  method->pdata = nullptr;
```

Comment:
why this change?  what's wrong with NULL?

Reply:
Technically speaking `NULL` is a character with the value 0, `nullptr` is a pointer. Having a pointer vs. non-pointer missmatch with `nullptr` will not create a compiler error or warning (depending on compiler and settings) which is helpful to detect incorrect use, while `NULL` will not. We are going to replace those gradually throughout the code. ... or I can do one big pull request that just changes this globally, if you prefer. This is one of the features of C++11 that is helpful to detect incorrect use at compile time. another would be to use "override" to detect unintentionally overloaded virtual functions.

Reply:
grrr had to update the message on github. one "not" too many.